docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+          </configurable-bundle-template-create>
+        </pages>
+      </configurable-bundle-templates>
+    </pages>
+  </product>
+</config>
+```
+
+## Permission Plugin
+
+```
+[Organization]
+├── Zed
+│   └── [Module]
+│       └── Communication
+│           └── Plugin
+│               └── [ConsumerModule]
+│                   └── [PluginName]Plugin.php
+├── Yves
+│   └── [Module]
+│       └── Plugin
+│           └── [ConsumerModule]
+│               └── [PluginName]Plugin.php
+├── Glue
+│   └── [Module]
+│       └── Plugin
+│           └── [ConsumerModule]
+│               └── [PluginName]Plugin.php    
+└── Client
+    └── [Module]        
+        └── Plugin
+            └── [ConsumerModule]
+                └── [PluginName]Plugin.php        
+```
+
+**Description**
+
+`Permission Plugins` are a way to put a scope on the usage of an application during a request lifecycle.
+
+**Conventions**
+
+- `Permission Plugins` MUST implement `\Spryker\Shared\PermissionExtension\Dependency\Plugin\PermissionPluginInterface`.
+- `Permission Plugins` MUST adhere to [Plugin](#plugin) conventions.
+
+**Guidelines**
+- When using one of the traits in a [Model](#model), use a `protected` constant string to reference the `Permission Plugin` that will be used in the [Model](#model). As an annotation for this string you should add `@uses` annotation to the `PermissionPlugin::KEY` string that you will compare against (see in example below).
+- The `\Spryker\[Application]\Kernel\PermissionAwareTrait` trait allows the [Model](#model) in the corresponding [application](#applications) to check if a permission is granted to an application user.
+
+**Example**
+
+```php
+class QuotePermissionChecker implements QuotePermissionCheckerInterface
+{
+use PermissionAwareTrait;
+
+    /**
+     * @uses \Spryker\Client\SharedCart\Plugin\ReadSharedCartPermissionPlugin::KEY
+     */
+    protected const PERMISSION_PLUGIN_KEY_READ_SHARED_CART = 'ReadSharedCartPermissionPlugin';
+
+    public function checkQuoteReadPermission(QuoteTransfer $quoteTransfer): bool
+    {
+        ...
+      return $this->can(
+        static::PERMISSION_PLUGIN_KEY_READ_SHARED_CART,
+        $quoteTransfer->getCustomer()->getCompanyUserTransfer()->getIdCompanyUser(),
+        $quoteTransfer->getIdQuote(),
+      );
+    }
+```
+
+## Persistence Schema
+
+```
+[Organization]
+└── Zed
+    └── [Module]
+        └── Persistence
+            └── Propel
+                └── Schema
+                    └── spy_[domain_name].schema.xml
+```
+
+**Description**
+
+The schema file defines the module's tables and columns (see [Propel Documentation - Schema XML](#https://propelorm.org/documentation/reference/schema.html)). Schema files are organized into business `domains`, each representing a module overarching group that encapsulates related domain entities.
+
+**Conventions**
+
+- Table name MUST follow the format `spy_[domain_entity_name]` (eg: `spy_customer_address`, `spy_customer_address_book`).
```

Comment:
`spy_[domain_entity_name]` It's not true for the storage, search or relation (spy_cms_slot_to_cms_slot_template)